docs/design/Versions and Releases.md

```diff
@@ -1,51 +1,107 @@
+in any way other than the reported test results being more correct. Small interface changes are permitted, as long as they
+are of such a nature that they do not affect current users. That is, the interface after a patch version upgrade should be
+a strict superset of the interface as it was before. Adding new result messages is explicitly permitted at this level, and
+external code using Zonemaster must always be prepared to handle messages it has not seen before.
 
 ## Development Releases
 
-Interim releases for development use (that is, intended for developers working on the versioned component itself rather than developers using the component) will use the CPAN convention of appending an underscore followed by a number. Perl's own version parsing and handling routines will treat a version on this form as comparing in between the version number without the underscore part and the version number before it. That is, all versions of the form 1.0.1\_nn compare as being between 1.0.1 and 1.0.0. Or, to look at it slightly differently, all 1.0.1\_nn versions are steps towards what is going to be 1.0.1.
+Interim releases for development use (that is, intended for developers working on the versioned component itself rather 
+than developers using the component) will use the CPAN convention of appending an underscore followed by a number. Perl's
+own version parsing and handling routines will treat a version on this form as comparing in between the version number
+without the underscore part and the version number before it. That is, all versions of the form 1.0.1\_nn compare as being
+between 1.0.1 and 1.0.0. Or, to look at it slightly differently, all 1.0.1\_nn versions are steps towards what is going to
+be 1.0.1.
 
 # Release Handling
 
 ## Development Releases
 
 Issued as needed.
 
-## Minor Versions
+## Patch Versions
 
-Should be easy to do, in order to encourage fixing problems quickly. All unit tests must pass, and all changes must be properly documented.
+The goal is that it should be easi to do patch releases, in order to encourage fixing problems quickly. All unit tests 
+must pass, and all changes must be properly documented. Besides that, enough testing should be completed before a new
+version is released.
 
-## Major Versions
+## Minor Versions
 
-These should be done with more care. Not only must all unit tests pass and everything be documented, but release candidates should be produced and known major users of the component should be encouraged to test the release candidates.
+These should be done with more care. Not only must all unit tests pass and everything be documented, but release 
+candidates should be produced and known major users of the component should be encouraged to test the release 
+candidates.
 
-## New Epoch
+## Major version
 
-If this happens, it will likely be treated as if it was an entirely new product.
+When this happens, it must be clearly documented and all parts of Zonemaster must work together.
```

Comment:
Automated tests for new functionality, integration testing between components and release candidates?

Reply:
Updated.